src/azurefluent/Model/MethodGroupJvaf.cs

```diff
@@ -104,34 +96,77 @@ private void DiscoverAllSupportedInterfaces()
-                var listMethod = this.Methods.FirstOrDefault(x => StringComparer.OrdinalIgnoreCase.Equals(x.Name, WellKnowMethodNames.List));
-
                 supportedInterfaces.Add($"{InnerSupportsListing}<{((ResponseJva)listMethod.ReturnType).SequenceElementTypeString}>");
                 interfacesToImport.Add($"{packageName}.{InnerSupportsListing}");
             }
         }
 
-        private static bool Take2RequiredParameters(Method method)
+        private static Method FindGetMethod(IEnumerable<Method> methods)
+        {
+            Method result = FindFirstMethodByName(methods, WellKnownMethodNames.GetByResourceGroup);
+            return result != null && HasNonClientNonConstantRequiredParameters(result, 2) ? result : null;
+        }
+
+        private static Method FindDeleteMethod(IEnumerable<Method> methods)
+        {
+            Method deleteMethod = FindFirstMethodByName(methods, WellKnownMethodNames.Delete);
+            return deleteMethod != null && HasNonClientNonConstantRequiredParameters(deleteMethod, 2) ? deleteMethod : null;
+        }
+
+        private static Method FindListMethod(IEnumerable<Method> methods)
+        {
+            Method result = null;
+
+            Method list = FindFirstMethodByName(methods, WellKnownMethodNames.List);
+            if (list != null && HasNonClientNonConstantRequiredParameters(list, 0))
+            {
+                Method listByResourceGroup = FindFirstMethodByName(methods, WellKnownMethodNames.ListByResourceGroup);
+                if (listByResourceGroup != null && HasNonClientNonConstantRequiredParameters(listByResourceGroup, 1))
+                {
+                    string listReturnType = GetSequenceElementTypeString(list);
+                    string listByResourceGroupReturnType = GetSequenceElementTypeString(listByResourceGroup);
+                    if (stringComparer.Equals(listReturnType, listByResourceGroupReturnType))
+                    {
+                        result = list;
+                    }
+                }
+            }
+
+            return result;
+        }
+
+        private static String GetSequenceElementTypeString(Method method)
+        {
+            return ((ResponseJva)method.ReturnType).SequenceElementTypeString;
+        }
+
+        private static Method FindFirstMethodByName(IEnumerable<Method> methods, String methodName)
+        {
+            return methods.FirstOrDefault(method => stringComparer.Equals(method.Name, methodName));
+        }
+
+        private static bool HasNonClientNonConstantRequiredParameters(Method method, int requiredParameterCount)
         {
             // When parameters are optional we generate more methods.
-            return method.Parameters.Count(x => !x.IsClientProperty && !x.IsConstant && x.IsRequired) == 2;
+            return method.Parameters.Count(x => !x.IsClientProperty && !x.IsConstant && x.IsRequired) == requiredParameterCount;

```

Comment:
What does it mean for something to be a client property? Does that mean the parameter to the REST API method is pulled from an instance variable and not passed as a method argument?

It's weird that 2 used to be a magic number here, and I'm wondering how/whether this is affecting the code gen now.